constellations/gemini-user/src/announcements/announce.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+require("./announce.sass");
+import React from "react";
+import API from "../utils/api.js";
+import { keys, sortBy, groupBy } from "lodash";
+import moment from "moment";
+
+export class AnnouncementGroup extends React.Component {
+    render() {
+        let a = this.props.announcements;
+        var postedAt = moment(this.props.postedAt).format('l');
+        let announcements = a.map((announcement, index) => (
+            <li key={announcement.id}>
+                <p>{announcement.message}</p>
+            </li>
+        ));
+
+        return (
+            <div className="announcement-card">
+                <h2>{postedAt}</h2>
+                <ul>{announcements}</ul>
+            </div>
+        );
+    }
+}
+
+export class AnnouncePage extends React.Component {
+    state = {
+        announcementList: [],
+    };
+    componentDidMount() {
+        console.log("api attempt ");
+        API.get("api/announcements/all").then((res) => {
+            const list = res.data;
+            console.log("api success!");
+            this.setState({ announcementList: list });
+        });
+    }
+    render() {
+        const announcements = this.state.announcementList;
+        let sorted = groupBy(announcements, (a) => a.postedAt.substring(0, 10));
```

Comment:
using substring on a date feels uncomfortable... especially because there are so many different date formats. I think we should be using `moment()` here and find a way to group announcements that have been posted on the same day. Moment has a `dayOfYear()` function which you can use to group announcements that have been announced on the same day?